app/models/host/managed.rb

```diff
@@ -314,12 +314,20 @@ def handle_realm
     end
   end
 
+  def disk_layout_template
+    if disk.present?
+      { name: 'Custom disk layout', template: disk }
+    elsif ptable.present?
+      { name: ptable.name, template: ptable.layout }
+    end
+  end
+
   # returns the host correct disk layout, custom or common
   def diskLayout
+    raise 'Neither disk nor partition table defined for host' unless disk_layout_template
```

Comment:
Shouldn't we raise a `ForemanException` (or some other meaningful exception) rather then a `RuntimeError` in these cases?
Also, do we `rescue` these exceptions anywhere?

Reply:
I suppose the exception is capture when rendering the provisioning template (similarly as other exceptions are), What about `Foreman::Renderer::RenderingError`?